Formula/synergy-core.rb

```diff
@@ -0,0 +1,351 @@
+class SynergyCore < Formula
+  desc "Synergy, the keyboard and mouse sharing tool"
+  homepage "https://symless.com/synergy"
+  url "https://github.com/symless/synergy-core/archive/refs/tags/1.14.3.5-stable.tar.gz"
+  sha256 "9259fc8aa02610121932cd5887c4d878beb08b2f7190e834c0584d53b36f5f9b"
+
+  # The synergy-core/LICENSE file contains the following preamble:
+  #    This program is released under the GPL with the additional exemption
+  #    that compiling, linking, and/or using OpenSSL is allowed.
+  # This preamble is followed by the text of the GPL-2.0.
+  #
+  # The list of SPDX license exceptions does not contain an exception corresponding
+  # to synergy-core's special OpenSSL exception. The openvpn-openssl-exception
+  # exception is very similar, but it is not the same, so it would be inappropriate
+  # to use it here.
+  #
+  # Unfortunately, writing an exception like this:
+  #     with: :cannot_represent
+  # causes `brew audit` to raise the following error:
+  #     Formula synergy-core contains invalid or deprecated SPDX license exceptions: [:cannot_represent].
+  # As a result, it appears we cannot mention the exception in the `license` definition.
+  license "GPL-2.0-only"
+
+  head "https://github.com/symless/synergy-core.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "openssl@3"
+  depends_on "pugixml"
+  depends_on "qt@5"
+
+  on_linux do
+    depends_on "pkgconfig" => :build
+    depends_on "gdk-pixbuf"
+    depends_on "glib"
+    depends_on "libnotify"
+    depends_on "libxkbfile"
+  end
+
+  # Build against our pugixml rather than the version included in the
+  # synergy-core repository. This is particularly necessary because the
+  # build process for synergy-core attempts to download the source code of
+  # pugixml and build it. This is not consistent with how things are done
+  # in brew, but it appears to be an intentional decision by upstream.
+  #
+  # Also, synergy-core's attempt to download the source code of pugixml
+  # works only if a `.git` directory exists in the source tree. Unless we're
+  # building with a git clone (rather than a release tarball), this doesn't
+  # even work correctly.
+  #
+  # The changelog for v1.13.0-stable includes the following item:
+  #   "[A]void using brew for build"
+  # As a result, it appears that this patch will not be accepted upstream.
+  #
+  # However, we still need to make this change to build this program, so
+  # this patch is unavoidable for now.
+  patch :DATA
```

Comment:
We still need this to be submitted upstream to consider accepting the formula

Reply:
Done.